### PR TITLE
Update SGX SDK to 2.20

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #36** <a href="https://app.graphite.dev/github/pr/mobilecoinofficial/docker-rust-sgx-base/36" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinofficial/docker-rust-sgx-base/36?utm_source=stack-comment).

File: Dockerfile [hunk 1]
```diff
@@ -95,8 +95,12 @@ ENV GOPATH=/opt/go/
 ENV PATH="/usr/local/go/bin:$GOPATH/bin:$PATH"
 RUN mkdir -p "${GOPATH}"
 
+ENV NODE_MAJOR=18
+
 # Add sources for nodejs and install it and other helpers from apt.
-RUN curl -LsSf https://deb.nodesource.com/setup_18.x | bash -s \
+RUN mkdir -p /etc/apt/keyrings \
```
Comment (nick-mobilecoin) [inline Dockerfile:101]: See, https://github.com/nodesource/distributions#new-update-%EF%B8%8F

The previous script put a 60 second delay in building the image, to catch ones attention to the deprecation.

File: Dockerfile [hunk 2]
```diff
@@ -45,16 +45,16 @@ RUN  ln -fs /usr/share/zoneinfo/Etc/UTC /etc/localtime \
...
   && chmod +x ./sgx.bin \
   && ./sgx.bin --prefix=/opt/intel \
   && rm ./sgx.bin
 
 # Install DCAP libraries
-ARG DCAP_VERSION=1.16.100.2-focal1
+ARG DCAP_VERSION=1.17.100.4-focal1
 RUN mkdir -p /etc/apt/keyrings \
-  && wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | gpg --dearmor | tee /etc/apt/keyrings/intel-sgx.gpg > /dev/null \
+  && wget -qO - https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | gpg --dearmor -o /etc/apt/keyrings/intel-sgx.gpg \
```
Comment (nick-mobilecoin) [inline Dockerfile:57]: Changed to '-o' as it was what the node instructions were using and seemed cleaner than tee

File: Dockerfile [hunk 3]
```diff
@@ -45,16 +45,16 @@ RUN  ln -fs /usr/share/zoneinfo/Etc/UTC /etc/localtime \
   && rm -r /var/lib/apt/lists
 
 # Install SGX
-ARG SGX_URL=https://download.01.org/intel-sgx/sgx-linux/2.19/distro/ubuntu20.04-server/sgx_linux_x64_sdk_2.19.100.3.bin
+ARG SGX_URL=https://download.01.org/intel-sgx/sgx-linux/2.20/distro/ubuntu20.04-server/sgx_linux_x64_sdk_2.20.100.4.bin
```
Comment (nick-mobilecoin) [inline Dockerfile:48]: 2.21 is also out, but wanted to do one bump at a time to make it easier to walk back if we hit any hiccups

File: Dockerfile (hunk 1 above)
Comment (jgreat) [reply Dockerfile:101]: Do we need to upgrade the nodejs version.  I'm not exactly sure why we need nodejs at this point. 

The only thing I can think of is some older mirror validation tests in full-service.  I think all those have been superseded by python tests?

**Review by jgreat — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (nick-mobilecoin) [reply Dockerfile:101]: I'm going to leave this in for now and we'll create a follow on to track down if we're still using node.

**Pull request merged**